main.go

```diff
@@ -19,12 +21,27 @@ func main() {
 	}
 
 	diskStorage := &storage.DiskStorage{Path: tmpDir}
-	appender := &appender.Appender{
-		Storage:            diskStorage,
-		MaxEntriesPerChunk: maxEntriesPerChunk,
+	appenders := []*appender.Appender{
+		&appender.Appender{
+			Storage:            diskStorage,
+			MaxEntriesPerChunk: maxEntriesPerChunk,
+		},
+		&appender.Appender{
+			Storage:            diskStorage,
+			MaxEntriesPerChunk: maxEntriesPerChunk,
+		},
+		&appender.Appender{
+			Storage:            diskStorage,
+			MaxEntriesPerChunk: maxEntriesPerChunk,
+		},
 	}
+	discovery := &discovery.Discovery{Appenders: appenders}
+	ingester := &ingester.Ingester{Discovery: discovery}
 
 	for i := 0; i < 40; i++ {
-		appender.HandleRequest(fmt.Sprintf("entry-%d", i))
+		err := ingester.HandleIngest(fmt.Sprintf("entry-%d", i))
+		if err != nil {
+			log.Panicf(fmt.Sprintf("Failed to handle ingest, %v", err))
```

Comment:
Similar to the error above, the message here would be more helpful if it provided context like the loop variable

Reply:
Done.